### PR TITLE
[bashible] creating deckhouser user and group

### PR DESCRIPTION
**Pull request by borg-z**
## Description
Added a function `bb-create_user_and_group` to create a user and group with specified IDs. 

## Why do we need it, and what problem does it solve?
This function ensures that the specified user and group are created with the correct IDs. It also handles cases where the user or group already exists, preventing conflicts. This is necessary to maintain consistency across different environments and avoid issues with user and group IDs.

## Why do we need it in the patch release (if we do)?
Not necessarily.

## What is the expected result?
The expected result is that the `deckhouse` user and group are created with the IDs 64535 and 64535. If the user or group already exists, appropriate warning messages are logged. If the group ID is already taken, the existing group is used for creating the user.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries
```changes
section: candi
type: feature
summary: Added function to create user and group with specified IDs, logging warnings if they already exist or IDs are taken.
impact: Ensures consistent user and group creation across different environments.
impact_level: default

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: candi/bashible/bashbooster/30_user.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+bb-create_user_and_group() {
+    local username="$1"
+    local userid="$2"
+    local groupname="$3"
+    local groupid="$4"
+
+    if id "$username" &>/dev/null; then
```
Comment (RomanenkoDenys) [inline candi/bashible/bashbooster/30_user.sh:21]: what is &> ?

File: candi/bashible/bashbooster/30_user.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        if [ "$(id -u "$username")" -eq "$userid" ] && [ "$(id -g "$username")" -eq "$groupid" ]; then
+            bb-log-warning "User $username with UID $userid and GID $groupid already exists. No changes needed."
```
Comment (RomanenkoDenys) [inline candi/bashible/bashbooster/30_user.sh:23]:
```suggestion
           
```

File: candi/bashible/bashbooster/30_user.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            fi
+        fi
+
+        # Check if user ID is already exists
+        if getent passwd "$userid" > /dev/null 2>&1; then
+            bb-log-warning "User ID $userid is already exists"
+        else
+            useradd -m -u "$userid" -g "$groupname" "$username"
+        fi
+    fi
```
Comment (RomanenkoDenys) [inline candi/bashible/bashbooster/30_user.sh:53]:
```suggestion
uid="$(id -u "${username}")"
gid="$(id -g "${groupname}")"

if [ "$(id -u "${username}")" -eq "${userid}" ] && [ "$(id -g "${groupname}")" -eq "${groupid}" ]; then
  return
fi 


if [ -n $uid ] && [ -n $gid ]; then
 bb-log-warning "user or group already exists with different id: uid=${uid}, gid=${gid}"
 return
}

useradd -m -u "$userid" -g "$groupname" "$username"
```

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some little nits

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/gcp'...

**Comment by name212** [comment]
/e2e/abort gcp;WithoutNAT;containerd;1.25 9584634962 9584634962-1-con-1-25 /sys/deckhouse-oss/install:pr8595 user@35.234.91.71

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/gcp'...

**Review by name212 — APPROVED** [review]

**Pull request merged**